### ISSUE TITLE
Error 0x8024400A - SOAPCLIENT_PARSE_ERROR

### ISSUE THREAD

**Issue by polybiusproxy** [kirb/LegacyUpdate#20]
Hi,

I am having this "SOAPCLIENT_PARSE_ERROR" when trying to search for updates.
![image](https://user-images.githubusercontent.com/47796739/204125360-d512520e-389c-4a1e-b0ec-b407b9239381.png)

I have everything set up correctly, and I don't really know a fix to this.

**Comment by thomasandersonthe3RD** [comment]
same

**Comment by bolabola6789** [comment]
Same issue here, running on XP Professional SP3 32-bit

Specs: 
Intel Core i5-3470
NVIDIA GeForce GT730
10GB of RAM (3.47GB recognized due to 32-bit limitations)

**Comment by thomasandersonthe3RD** [comment]
> Same issue here, running on XP Professional SP3 32-bit
> 
> Specs: Intel Core i5-3470 NVIDIA GeForce GT730 10GB of RAM (3.47GB recognized due to 32-bit limitations)

hey i know how to fix it and where the problem come from
you have a corrupted installation of windows xp or
you algredy modify  windows update with another update fixer or somthing like that
in either case you will need to reinstall for me it worked

**Comment by bolabola6789** [comment]
Nevermind, I got it worked. I just retry a few more times

Vào 23:58, Th 7, 17 thg 12, 2022 thomasandersonthe3RD <
***@***.***> đã viết:

**Comment by thomasandersonthe3RD** [comment]
> 

oh cool! 
for me i needed to reinstall it :(

**Comment by Sm0keySa1m0n** [comment]
Getting this on Windows 2000

**Comment by kirb** [comment]
Confirmed this happens when the computer has previously been configured for a WSUS server. Will take another stab at fixing this then. 

If this happens to you, try these steps to clean up the WSUS config on your machine so it can use public servers again: https://github.com/kirb/LegacyUpdate/issues/12#issuecomment-1319401341

**Comment by arjanvlek** [comment]
I've had the same issue. I dicussed with @kirb via private Twitter message and it seemed to be caused by previously having a *private* WSUS server configured on both my machines.

As just mentioned above, the steps from https://github.com/kirb/LegacyUpdate/issues/12 have fixed it for me.

I'd like to mention that the initial check for updates after performing the steps takes a **very long** time:

- It took almost 30 minutes on my XP laptop, which has a fast-for-XP AMD 64 3400+ processor and 1.5gb of RAM. 
- It took well over *75 minutes* on my Vista UMPC, which is a Sony UX380N. 

The big difference in time is likely caused by running Vista in combination with the very slow hard drive that is inside the Sony UMPC.

**Comment by kirb** [comment]
Sorry @arjanvlek, didn’t know if you wanted to be mentioned. Credit goes to you for helping me make the connection to this error coming from the WSUS registration step, thanks!

**Comment by kirb** [comment]
I think this is now fixed (for real this time).

When talking to a server, the server sends a cookie to the Windows Update Agent on your machine, so it can keep track of the computer and the updates its agent is aware of, so it doesn’t send the same updates to it more than once. When switching between completely different servers, as would be the case switching between a private WSUS server and the public Microsoft server (which Legacy Update proxies), the server is supposed to send an “invalid cookie” error. The state from the old server is no longer relevant, and the new server won’t be aware of the cookie anyway, so this error is a signal that the agent needs to clean up and start over.

One of the things the agent requests from the server is its configuration. WSUS servers’ configuration indicates that they require the agent to use “SimpleAuth”, while the public server (and Legacy Update) presents itself as “Anonymous”. Due to a mistake on my part, the “invalid cookie” response was being returned as HTTP 200 OK, rather than HTTP 500 Internal Server Error, which I must have missed because 500 is a strange way to indicate a *client* error (that’s supposed to be HTTP 400 Bad Request!). So the agent tries to check in with the SimpleAuthWebService on the Legacy Update server, which doesn’t exist. Or, it tries to continue business as usual, asking for more updates, which returns an “invalid cookie” response the agent can’t understand due to the incorrect status code – hence the parse error.

Now, I can confirm that it correctly switches servers, although only after returning a strange 800A0005 error, which corresponds to “the syntax in the function call was incorrect”. No idea why, but after checking for updates again, it works.

**Comment by kirb** [comment]
Updated the server to display a custom error message when 800A0005 is encountered: “Legacy Update has successfully deregistered your computer from a Windows Server Update Services (WSUS) server. Check for updates again to continue.”
